### PR TITLE
fix code smells

### PR DESCRIPTION
**Pull request by XenoAmess**

### CODE REVIEW

File: tool/src/org/antlr/v4/tool/ErrorManager.java [hunk 1]
```diff
@@ -193,15 +193,13 @@ private static StackTraceElement getLastNonErrorManagerCodeLocation(Throwable e)
 	public void emit(ErrorType etype, ANTLRMessage msg) {
 		switch ( etype.severity ) {
 			case WARNING_ONE_OFF:
+			case ERROR_ONE_OFF:
```
Comment (sharwell) [inline tool/src/org/antlr/v4/tool/ErrorManager.java:196]: ❗ this is not correct

Comment (XenoAmess) [reply tool/src/org/antlr/v4/tool/ErrorManager.java]: @sharwell Sorry, my bad.

File: runtime-testsuite/test/org/antlr/v4/test/runtime/csharp/BaseCSharpTest.java [hunk 2]
```diff
@@ -245,8 +245,7 @@ public String execLexer(String grammarFileName,
 	Set<String> sourceFiles = new HashSet<String>();
 
 	private void addSourceFiles(String ... files) {
-		for(String file : files)
-			this.sourceFiles.add(file);
+		this.sourceFiles.addAll(Arrays.asList(files));
```
Comment (NathanQingyangXu) [inline runtime-testsuite/test/org/antlr/v4/test/runtime/csharp/BaseCSharpTest.java:248]: There might be better optimization here:
```
Collections.addAll(this.sourceFiles, files);
```
As per Javadoc, its function remains the same, but it is likely to run significantly faster under most implementations

Comment (XenoAmess) [reply runtime-testsuite/test/org/antlr/v4/test/runtime/csharp/BaseCSharpTest.java:248]: @NathanQingyangXu 
Hi.
After further test I notice you are right, this is a wrong optimize, it runs slightly slower in HashSet implementation.
Should revoke to original.

**Comment by parrt** [comment]
Hi.  I agree with Sam that it's too big of a change. Sorry.

**Pull request closed without merging**